feat(admin-dashboard): created recent orders functionality

rohansen856 opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #89 .
2. This PR does the following: created recent orders details card with data fetching.

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
- [x] The PR is made from a branch that's **not** called "main/master".

## Proof that changes are correct
![Screenshot 2024-06-16 232750](https://github.com/bsoc-bitbyte/busify/assets/108976786/38a48c1a-d7d5-43e7-8e63-d2e41028e401)
![Screenshot 2024-06-16 232726](https://github.com/bsoc-bitbyte/busify/assets/108976786/8b58a883-c90c-4f84-9e10-1034ce50b226)

responsiveness:
![Screenshot 2024-06-16 232508](https://github.com/bsoc-bitbyte/busify/assets/108976786/152250f2-d22a-455a-9a51-8daea3bf9984)
![Screenshot 2024-06-16 232534](https://github.com/bsoc-bitbyte/busify/assets/108976786/24c752c6-85c6-46d5-b347-07bb7ff0d097)
![Screenshot 2024-06-16 232523](https://github.com/bsoc-bitbyte/busify/assets/108976786/cbbf4422-5506-4686-8d10-f7839a62614b)



<!--
Add videos/screenshots of the user-facing interface to demonstrate that the changes
made in this PR work correctly. If this PR is for a developer-facing feature,
provide the videos/screenshots of developer-facing interface.

Please also include videos/screenshots of the developer tools
browser console, so that we can be sure that there are no console errors.

If the changes in your PRs are autogenerated via a script and you cannot provide proof
for the changes then please leave a comment "No proof of changes needed because {{Reason}}"
and remove all the sections below.
-->

works done in this pr:
### frontend
1. created a list view of the recent orders (orders done inn the same day).
2. created order details card, viewable when clicked `View Details`

### backend
1. modified the `orders.service.ts` file to add the logic to get the data in required format
2. modified the `orders.controller.ts` file to add the route handler to get the details.
3. fixed the seed file to push the data in better format.


## PR Pointers

- If you need a review or an answer to a question, and don't have permissions to assign people, **leave a comment** like the following: "{{Question/comment}} @{{reviewer_username}} PTAL".
- Never force push. If you do, your PR will be closed.

RISHIKESHk07 commented:
@rohansen856 we will be having relationship between tickets and orders as well , it make it easier for the other issue for data  fetching , please make changes according to this schema 
![WhatsApp Image 2024-06-17 at 9 46 09 AM](https://github.com/bsoc-bitbyte/busify/assets/117551967/0b5f9a33-42f8-421c-8e50-d3a3f2da2d8f)

rohansen856 commented:
@RISHIKESHk07 i will fix the issue and update the schema with the correct relationships.

rohansen856 commented:
@RISHIKESHk07  @Pratik2026  i have updated the `schema.prisma` file to make relations between orders, tickets and schedule schma. also updated the data fetching method in `order.service.ts` in my latest pr.
the current relationship is:
```prisma
model Schedule {
    id            String   @id
    busNumber     String
    bus           Bus      @relation(fields: [busNumber], references: [number])
    checkpoints   String[]
    from          String
    to            String
    departureTime String
    days          String[]
    ticketPrice   Int
    orders        Order[]
}

model Ticket {
    id             Int @id @default(autoincrement())
    orderId        String @unique
    passengerEmail String[]
    order          Order @relation(fields: [orderId], references: [id])
}

model Order {
    id         String     @id
    userId     String
    scheduleId String
    status     String     @default("pending") // how to make this enum
    amount     Int
    attempts   Int        @default(0)
    receipt    String     @unique
    createdAt  DateTime   @default(now())
    user       Users      @relation(fields: [userId], references: [id])
    ticket     Ticket?
    schedule   Schedule  @relation(fields: [scheduleId], references: [id])
}
```


 thhank you.

RISHIKESHk07 commented:
@Pratik2026 LGTM  codewise , any changes as per you ?

RISHIKESHk07 commented on frontend/src/components/AdminPanel/OrderDetailsCard/index.tsx:
```diff
@@ -0,0 +1,253 @@
+import { Typography, styled, Box, useTheme } from '@mui/material';
+import  Clear  from "@mui/icons-material/clear"
```
@rohansen856  double check this  line @mui/icons-material/clear"

rohansen856 replied:
I will fix the issue with the formatting and add the semicolon(`;`) in this line.

RISHIKESHk07 replied:
@rohansen856 nah the probelm is the its "@mui/icons-material/Clear" , on my end i see it giving a error for ...../clear

RISHIKESHk07 reviewed (changes requested):
@rohansen856 did you verify if page works when we add a ticket through the razorpay UI (dummy order) instead of seeding the data ? i tried it here (Left: Busify & Right:Ptisma Studio) , there is a ticket created but does not appear in the admin page , maybe its a date issue , you can test razorpay UPI with success@razorpay
![image](https://github.com/bsoc-bitbyte/busify/assets/117551967/28d8047b-6b0c-465d-a2c7-63e2cbef5841)

rohansen856 commented:
> @rohansen856 did you verify if page works when we add a ticket through the razorpay UI (dummy order) instead of seeding the data ? i tried it here (Left: Busify & Right:Ptisma Studio) , there is a ticket created but does not appear in the admin page , maybe its a date issue , you can test razorpay UPI with success@razorpay ![image](https://private-user-images.githubusercontent.com/117551967/341132810-28d8047b-6b0c-465d-a2c7-63e2cbef5841.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg4MTI4NTMsIm5iZiI6MTcxODgxMjU1MywicGF0aCI6Ii8xMTc1NTE5NjcvMzQxMTMyODEwLTI4ZDgwNDdiLTZiMGMtNDY1ZC1hMmM3LTYzZTJjYmVmNTg0MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxOVQxNTU1NTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01YzQ4ZGM2NDU1NzlkODU3OTY4MWQ4MWIwZDJhNjg5NWU3ZTgxNjQzYTY5NTg4N2Q1MDY3M2Q3Y2QzMjg1ZWJhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.MvRlD5SLb_USwyWG0nkdDINP8ED2ilybcxTi2tKVrx4)

@RISHIKESHk07 i will double check the payments issue with razorpay, and also check any issues with the data.

rohansen856 commented:
@RISHIKESHk07 i have solved the following issues in the latest pr:
1. updated the prisma `Ticket` schema with the discussed changes.
2. updated the order of database seeding in `prisma/seed.mjs`.
3. updated `orders.service.ts` to fix the issue with the date object in razorpay payment datetime.
4. fixed the frontend issue with parsing the datetime by encapsulation the field in `new Date()`.
5. fixed the issue with formatting in `frontend/src/components/AdminPanel/OrderDetailsCard/index.tsx`.
proof of work:

https://github.com/bsoc-bitbyte/busify/assets/108976786/e01cbc5c-5195-46ae-b71b-cd60db4765d8


https://github.com/bsoc-bitbyte/busify/assets/108976786/57c85be1-8913-4363-9193-5273edd08616

thank you.
